src/v/cluster/rm_stm.cc

```diff
@@ -1706,19 +1707,32 @@ ss::future<> rm_stm::do_try_abort_old_tx(model::producer_identity pid) {
 
     _mem_state.expected.erase(pid);
 
+    model::tx_seq transaction_seq;
     std::optional<prepare_marker> marker;
-    auto prepare_it = _mem_state.preparing.find(pid);
-    if (prepare_it != _mem_state.preparing.end()) {
-        marker = prepare_it->second;
-    }
-    prepare_it = _log_state.prepared.find(pid);
-    if (prepare_it != _log_state.prepared.end()) {
-        marker = prepare_it->second;
+
+    if (is_transaction_ga()) {
```

Comment:
is there a more descriptive name than "transaction ga"? something that communicates the change in semantics? it's ok if there isn't--perhaps nothing fits. but generally this isn't ideal because someone reading this code has to go track down what exactly GA introduced. And the common phrasing "General Availability" usually means that something is now stable not that semantics changed in a significant way.

Reply:
this was discussed in a different PR. conclusion was to leave the name but link to relevant information for understanding what the flag really means.